consensus/src/messages.rs

```diff
@@ -204,3 +216,109 @@ impl PartialEq for QC {
         self.hash == other.hash && self.round == other.round
     }
 }
+
+#[derive(Clone, Serialize, Deserialize)]
+pub struct Timeout {
+    pub high_qc: QC,
+    pub round: RoundNumber,
```

Comment:
For efficiency the timeout signature should be over the round the node is timing out and the round inside the high_qc. This way the leader can forward all O(n) signatures and a single high_qc inside the TC.

Reply:
Yes, this is exactly what we do: https://github.com/asonnino/hotstuff/blob/35c6a9985316c1898b53f0d0c37c56a823f30d77/consensus/src/messages.rs#L266